app/javascript/components/explore/ExploreDisplayTabs.jsx

```diff
@@ -230,8 +230,19 @@ export default function ExploreDisplayTabs({
   const isCorrelatedScatter = enabledTabs.includes('correlatedScatter')
 
 
-  // if the exploreParams update need to reset the initial cell facets
+  // If clustering or annotation changes, then update facets shown for cell filtering
   useEffect(() => {
+    if (!exploreInfo) {return}
+    if (exploreInfo.skipFetchFacets) {
+      // The loadStudyData in ExploreView updates exploreParams _twice_ upon
```

Comment:
I think this highlights a lurking issue that we have throughout the explore UX - we tie a lot of processing/rendering off of `useEffect` calls that look for changes to `exploreParams`.  I wonder if consolidation of these calls into a more centralized function would help.